subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/TaskDetailsFactory.java

```diff
@@ -17,17 +17,18 @@
 
 import org.gradle.api.Project;
 import org.gradle.api.Task;
+import org.gradle.api.internal.project.ProjectInternal;
 import org.gradle.util.Path;
 
 import java.util.Set;
 
 public class TaskDetailsFactory {
-    private final Set<Project> projects;
+    private final Set<? extends ProjectInternal> projects;
     private final Project project;
 
     public TaskDetailsFactory(Project project) {
         this.project = project;
-        this.projects = project.getAllprojects();
+        this.projects = ((ProjectInternal)project).getAllprojectsUnchecked();
```

Comment:
❌  The `TaskDetailsFactory` seems to be doing unsafe cross-project configuration. We should either find a way to do it safely, or to keep the unsafe access, so that violations would surface, if anyone is using this API.